// @privy-io/react-auth/dist/cjs/SelectSourceAsset-DL13gM45.js
"use strict";var e=require("./use-export-wallet-Bdq329pR.js"),r=require("react/jsx-runtime"),t=require("lucide-react"),o=require("react"),n=require("styled-components"),l=require("./index-DDHMB6qK.js"),s=require("./ScreenLayout-CLeA5Eh7.js");let i=n.styled.span`
  background-color: var(--privy-color-background);
  width: 100%;
  text-align: center;
  border: none;
  font-kerning: none;
  font-feature-settings: 'calt' off;
  display: flex;
  justify-content: center;
  align-items: flex-start;
  cursor: pointer;

  &:focus {
    outline: none !important;
    border: none !important;
    box-shadow: none !important;
  }

  && {
    color: var(--privy-color-foreground);
    font-size: 3.75rem;
    font-style: normal;
    font-weight: 600;
    line-height: 5.375rem;
  }
`,a=n.styled.span`
  color: var(--privy-color-foreground);
  font-kerning: none;
  font-feature-settings: 'calt' off;
  font-size: 1rem;
  font-style: normal;
  font-weight: 600;
  line-height: 1.5rem;
  margin-top: 0.75rem;
`,d=n.styled.button`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: auto;
  gap: 0.5rem;
  border: 1px solid var(--privy-color-border-default);
  border-radius: var(--privy-border-radius-full);

  && {
    margin: auto;
    padding: 0.5rem 1rem;
  }
`,c=n.styled.div`
  svg {
    width: 1rem;
    height: 1rem;
    border-radius: var(--privy-border-radius-full);
    overflow: hidden;
  }
`,u=n.styled.span`
  color: var(--privy-color-foreground);
  font-kerning: none;
  font-feature-settings: 'calt' off;
  font-size: 0.875rem;
  font-style: normal;
  font-weight: 500;
  line-height: 1.375rem;
`,p=n.styled.div`
  color: var(--privy-color-foreground);

  svg {
    width: 1.25rem;
    height: 1.25rem;
  }
`;let f=n.styled.div`
  display: flex;
  flex-direction: column;
  gap: 0.75rem;
  width: 100%;
`,h=n.styled.button`
  border-color: var(--privy-color-border-default);
  border-width: 1px;
  border-radius: var(--privy-border-radius-mdlg);
  border-style: solid;
  display: flex;

  && {
    padding: 0.75rem 1rem;
  }
`,y=n.styled.div`
  display: flex;
  align-items: center;
  gap: 1rem;
  width: 100%;
`,m=n.styled.div`
  svg {
    width: 2.25rem;
    height: 2.25rem;
    border-radius: var(--privy-border-radius-full);
    overflow: hidden;
  }
`,g=n.styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  gap: 0.125rem;
`,v=n.styled.span`
  color: var(--privy-color-foreground);
  font-size: 0.875rem;
  font-weight: 600;
  line-height: 1.25rem;
`,x=n.styled.span`
  color: var(--privy-color-foreground-3);
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 1.125rem;
`;exports.AmountInput=({currency:e="usd",value:t,onChange:n,inputMode:s="decimal",autoFocus:d})=>{let[c,u]=o.useState("0"),p=o.useRef(null),f=t??c,h=l.SUPPORTED_CURRENCIES[e]?.symbol??"$",y=o.useCallback((e=>{let r=e.target.value,t=(r=r.replace(/[^\d.]/g,"")).split(".");t.length>2&&(r=t[0]+"."+t.slice(1).join("")),2===t.length&&t[1].length>2&&(r=`${t[0]}.${t[1].slice(0,2)}`),r.length>1&&"0"===r[0]&&"."!==r[1]&&(r=r.slice(1)),(""===r||"."===r)&&(r="0"),n?n(r):u(r)}),[n]),m=o.useCallback((e=>{!(["Delete","Backspace","Tab","Escape","Enter",".","ArrowLeft","ArrowRight","ArrowUp","ArrowDown","Home","End"].includes(e.key)||(e.ctrlKey||e.metaKey)&&["a","c","v","x"].includes(e.key.toLowerCase()))&&(e.key>="0"&&e.key<="9"||e.preventDefault())}),[]),g=o.useMemo((()=>(f.includes("."),f)),[f]);/*#__PURE__*/return r.jsxs(i,{onClick:()=>p.current?.focus(),children:[/*#__PURE__*/r.jsx(a,{children:h}),g,/*#__PURE__*/r.jsx("input",{ref:p,type:"text",inputMode:s,value:g,onChange:y,onKeyDown:m,autoFocus:d,placeholder:"0",style:{width:1,height:"1rem",opacity:0,alignSelf:"center",fontSize:"1rem"}}),/*#__PURE__*/r.jsx(a,{style:{opacity:0},children:h})]})},exports.CurrencySelector=({selectedAsset:e,onEditSourceAsset:o})=>{let{icon:n}=l.SUPPORTED_CURRENCIES[e];/*#__PURE__*/return r.jsxs(d,{onClick:o,children:[/*#__PURE__*/r.jsx(c,{children:n}),/*#__PURE__*/r.jsx(u,{children:e.toLocaleUpperCase()}),/*#__PURE__*/r.jsx(p,{children:/*#__PURE__*/r.jsx(t.ChevronDown,{})})]})},exports.SelectSourceAsset=({opts:e,isLoading:t,onSelectSource:o})=>/*#__PURE__*/r.jsx(s.ScreenLayout,{showClose:!1,showBack:!0,onBack:()=>o(e.source.selectedAsset),title:"Select currency",children:/*#__PURE__*/r.jsx(f,{children:e.source.assets.map((e=>{let{icon:n,name:s}=l.SUPPORTED_CURRENCIES[e];/*#__PURE__*/return r.jsx(h,{onClick:()=>o(e),disabled:t,children:/*#__PURE__*/r.jsxs(y,{children:[/*#__PURE__*/r.jsx(m,{children:n}),/*#__PURE__*/r.jsxs(g,{children:[/*#__PURE__*/r.jsx(v,{children:s}),/*#__PURE__*/r.jsx(x,{children:e.toLocaleUpperCase()})]})]})},e)}))})}),exports.poll=async({operation:r,until:t,delay:o,interval:n,attempts:l,signal:s})=>{let i,a,d=0;for(;d<l;){if(s?.aborted)return{status:"aborted",result:i,attempts:d,error:a};d++;try{if(a=void 0,i=await r(),t(i))return{status:"success",result:i,attempts:d};d<l&&await e.sleep(n)}catch(r){r instanceof Error&&(a=r),d<l&&await e.sleep(n)}}return{status:"max_attempts",result:i,attempts:d,error:a}};
